src/pynxtools/dataconverter/helpers.py

```diff
@@ -658,6 +658,26 @@ def convert_str_to_bool_safe(value: str) -> Optional[bool]:
     raise ValueError(f"Could not interpret string '{value}' as boolean.")
 
 
+def convert_int_to_float(value):
+    """
+    Converts int-like values to float, including values in arrays, and lists
+
+    Args:
+        value: The input value, which can be a single value, list, or numpy array.
+
+    Returns:
+        The input value with all int-like values converted to float.
+    """
+    if isinstance(value, int):
+        return float(value)
+    elif isinstance(value, list):
+        return [convert_int_to_float(v) for v in value]
```

Comment:
Not sure if this ever comes up, but just for the sake of being comprehensive, we can add support for sets and tuples.
```suggestion
    elif isinstance(value, list):
        return [convert_int_to_float(v) for v in value]
    elif isinstance(value, tuple):
        return tuple(convert_int_to_float(v) for v in value)
    elif isinstance(value, set):
        return {convert_int_to_float(v) for v in value}
```

Reply:
That's a good idea, I will add this